rust/cubesqlplanner/cubesqlplanner/src/plan/join.rs

```diff
@@ -192,6 +215,7 @@ pub enum JoinCondition {
     BaseJoinCondition(Rc<dyn BaseJoinCondition>),
     RegularRollingWindowJoinCondition(RegularRollingWindowJoinCondition),
     ToDateRollingWindowJoinCondition(ToDateRollingWindowJoinCondition),
+    RollingTotalJoinCondition(RollingTotalJoinCondition),
```

Comment:
Shouldn't all join conditions implement some kind of trait with `to_sql()` func so you can just call it on any condition?